src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -38,6 +44,12 @@ private sealed class EasyRequest : TaskCompletionSource<HttpResponseMessage>
             internal SendTransferState _sendTransferState;
             internal bool _isRedirect = false;
 
+            private SafeCallbackHandle _headerCallbackHandle;
+            private SafeCallbackHandle _sendCallbackHandle;
+            private SafeCallbackHandle _receiveBodyCallback;
+            private SafeCallbackHandle _seekCallbackHandle;
+            private SafeCallbackHandle _sslCtxCallbackHandle;
```

Comment:
One cheap option would be to just hold on to the IntPtrs instead of making SafeHandle objects. And then directly calling FreeCallbackHandle on each of the IntPtrs.